src/plugins/discover/public/embeddable/saved_search_embeddable.tsx

```diff
@@ -584,6 +586,9 @@ export class SavedSearchEmbeddable
     if (this.searchProps) {
       delete this.searchProps;
     }
+    if (this.node) {
+      unmountComponentAtNode(this.node);
+    }
```

Comment:
👍 for catching this, seems we didn't clean up?

Reply:
It were doing it in render function on each rerender, which looks wrong.